Client/Frontend/Login/LoginInfoViewController.swift

```diff
@@ -0,0 +1,380 @@
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch section {
+            case Section.information.rawValue:
+                return InfoItem.allCases.count
+            default:
+                return 1
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        switch indexPath.section {
+            case Section.information.rawValue:
+                return UX.informationRowHeight
+            case Section.createdDate.rawValue:
+                return UX.createdRowHeight
+            case Section.delete.rawValue:
+                return UX.standardItemHeight
+            default:
+                return UITableView.automaticDimension
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        switch indexPath.section {
+            case Section.delete.rawValue:
+                deleteLogin()
+            default:
+                if !isEditingFieldData {
+                    showActionMenu(for: indexPath)
+                }
+        }
+        
+        tableView.deselectRow(at: indexPath, animated: true)
+    }
+}
+
+// MARK: Actions
+
+extension LoginInfoViewController {
+    
+    @objc private func edit() {
+        isEditingFieldData = true
+        
+        navigationItem.rightBarButtonItem =
+            UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(doneEditing))
+
+        cellForItem(InfoItem.usernameItem)?.descriptionTextField.becomeFirstResponder()
+    }
+    
+    private func showActionMenu(for indexPath: IndexPath) {
+        if indexPath.section == Section.createdDate.rawValue || indexPath.section == Section.delete.rawValue {
+            return
+        }
+
+        guard let cell = tableView.cellForRow(at: indexPath) as? LoginInfoTableViewCell else {
+            return
+        }
+        cell.becomeFirstResponder()
+        
+        UIMenuController.shared.showMenu(from: tableView, rect: cell.frame)
+    }
+    
+    @objc private func doneEditing() {
+        isEditingFieldData = false
+        navigationItem.rightBarButtonItem =
+            UIBarButtonItem(barButtonSystemItem: .edit, target: self, action: #selector(edit))
+        
+        guard let username = usernameField?.text, let password = passwordField?.text,
+              username != loginEntry.username || password != loginEntry.password else {
+            return
+        }
+                
+        loginEntry.update(password: password, username: username)
+        
+        profile.logins.updateLoginByGUID(loginEntry.guid, new: loginEntry, significant: true).upon { [weak self] result in
+            DispatchQueue.main.async {
+                if result.isSuccess {
+                    self?.tableView.reloadData()
+                } else {
+                    log.error("Error while updating a login entry")
```

Comment:
Does `result` store any error code? if yes we can add it to the log

Reply:
Yeah it return a MaybeErrorType and it is an Error and StringConvertible

Fix at https://github.com/brave/brave-ios/pull/4124/commits/ded5065d24f539a7dee6bb8bcb40b93a2b70fe7b